plugin_ide.ui/src/com/github/rustdt/ide/ui/editor/RustEditorActionContributor.java

```diff
@@ -28,6 +47,88 @@ protected RustOpenDefinitionOperation createOpenDefinitionOperation(ITextEditor
 	
 	@Override
 	protected void registerOtherEditorHandlers() {
+		setupOpenType();

```

Comment:
> Besides that, I am wondering how the language agnostic Open Type dialog will work in the end. We could just call LangCore.getIndexManager() but probably there is a better solution in place.

Yeah that is fine actually. If it comes that elsewhere we need to get `RustIndexManager` instead of just `IndexManager`, an accessor method can be created in LangCore_Actual that provides that. For example, that is done in DDT, for the ToolManager and BundleModelManager singletons, there are accessors with language-specific return types:
https://github.com/DDT-IDE/DDT/blob/1b68c437fd680d0478c7b97fe43c926d4da902b7/plugin_ide.core/src/melnorme/lang/ide/core/LangCore_Actual.java#L57
It may look a bit odd at times, but in practice it works fine.